src/Hcal/HcalTrigPrimDigiProducer.cxx

```diff
@@ -0,0 +1,76 @@
+#include "Hcal/HcalTrigPrimDigiProducer.h"
+
+#include "Hcal/HcalTriggerGeometry.h"
+#include "Recon/Event/HgcrocDigiCollection.h"
+#include "Recon/Event/HgcrocTrigDigi.h"
+#include "Tools/HgcrocTriggerCalculations.h"
+
+namespace hcal {
+HcalTrigPrimDigiProducer::HcalTrigPrimDigiProducer(const std::string& name,
+                                                   framework::Process& process)
+    : Producer(name, process) {}
+
+void HcalTrigPrimDigiProducer::configure(framework::config::Parameters& ps) {
+  digiCollName_ = ps.getParameter<std::string>("digiCollName");
+  digiPassName_ = ps.getParameter<std::string>("digiPassName");
+  condObjName_ =
+      ps.getParameter<std::string>("condObjName", "HcalTrigPrimDigiConditions");
+}
+
+void HcalTrigPrimDigiProducer::produce(framework::Event& event) {
+  const HcalTriggerGeometry& geom = getCondition<HcalTriggerGeometry>(
+      HcalTriggerGeometry::CONDITIONS_OBJECT_NAME);
+
+  const ldmx::HgcrocDigiCollection& hcalDigis =
+      event.getObject<ldmx::HgcrocDigiCollection>(digiCollName_, digiPassName_);
+
+  // get the calibration object
+  const conditions::IntegerTableCondition& conditions =
+      getCondition<conditions::IntegerTableCondition>(condObjName_);
+
+  // construct the calculator...
+  ldmx::HgcrocTriggerCalculations calc(conditions);
```

Comment:
Need to discuss  w/ hcal digi experts what updates are required to this calculation.

Reply:
done